### PR TITLE
Document ScreenHandler

### PR DESCRIPTION
**Pull request by apple502j**
Documents `ScreenHandler` and other things. Draft for now, expect more docs on other classes.

### CODE REVIEW

File: mappings/net/minecraft/screen/NamedScreenHandlerFactory.mapping [hunk 1]
```diff
@@ -1,4 +1,11 @@
 CLASS net/minecraft/class_3908 net/minecraft/screen/NamedScreenHandlerFactory
+	COMMENT A screen handler factory with a name (title). This is passed to {@link
+	COMMENT net.minecraft.entity.player.PlayerEntity#openHandledScreen} to open a screen
+	COMMENT handler.
+	COMMENT
+	COMMENT <p>In vanilla, block entity instances implement this interface, allowing them to be used
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/NamedScreenHandlerFactory.mapping]:
```suggestion
	COMMENT <p>In vanilla, most block entity instances implement this interface, allowing them to be used
```

File: mappings/net/minecraft/screen/ScreenHandlerContext.mapping [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
-	COMMENT Screen handler contexts allow screen handlers to interact with the
-	COMMENT logical server's world safely.
+	COMMENT A screen handler context allows running code on the server side only. Screen
+	COMMENT handlers are designed to be used on both sides; any action modifying the world has
+	COMMENT to be wrapped in a call to the context. This guarantees that no casting error occurs
+	COMMENT inside the screen handler code.
+	COMMENT
+	COMMENT <p>A context with the world is passed to the screen handler on creation on the server.
+	COMMENT On the server, the context executes the function with the world and the position.
+	COMMENT On the client, the empty context is used.
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandlerContext.mapping]:
```suggestion
	COMMENT On the client, the {@linkplain #EMPTY empty context} is used.
```

File: mappings/net/minecraft/screen/ScreenHandlerType.mapping [hunk 3]
```diff
@@ -12,6 +24,10 @@ CLASS net/minecraft/class_3917 net/minecraft/screen/ScreenHandlerType
 		ARG 0 syncId
 		ARG 1 playerInventory
 	CLASS class_3918 Factory
+		COMMENT A functional interface that creates a screen handler instance on the client.
+		COMMENT
+		COMMENT <p>Screen handlers usually have a constructor that can be used as an implementation.
+		COMMENT See the note at {@link ScreenHandler}.
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandlerType.mapping]:
```suggestion
		COMMENT See the note on {@link ScreenHandler}.
```
"at" sounds a bit weird to me? Not sure if "on" is entirely correct either though 😅

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 4]
```diff
@@ -1,4 +1,128 @@
...
+	COMMENT <p>A subclass should have two constructors. One is for the server, and should take
+	COMMENT the {@code syncId} and inventories, property delegates, or contexts that are used.
+	COMMENT The {@link #syncId} is shared between the two sides. It is used to verify that a player
+	COMMENT has a specific screen (handler) open so that they can move items, for example.
+	COMMENT The inventories are used to back a slot so that any changes to a slot is reflected
+	COMMENT on the backing inventory, and vice versa. Property delegates and contexts bridge
+	COMMENT between the screen handler and other parts of the world; see above for more description.
+	COMMENT
+	COMMENT <p>The constructor should {@linkplain #addSlot add slots}, {@link #addProperties
+	COMMENT add properties from the delegate}, and store the property delegate and screen handler
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]:
```suggestion
	COMMENT add properties from delegates}, and store the property delegates and screen handler
```

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 5]
```diff
@@ -1,4 +1,128 @@
...
+	COMMENT context in the instance field.
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]:
```suggestion
	COMMENT context in the instance fields.
```

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 6]
```diff
@@ -1,4 +1,128 @@
...
+	COMMENT
+	COMMENT <p>The other constructor is for the client. There, the only parameters allowed are the
+	COMMENT {@code syncId} and the player inventory. This is because all other things are
+	COMMENT unavailable at creation time and synced later. This constructor should call the
+	COMMENT other constructor with {@linkplain net.minecraft.inventory.SimpleInventory
+	COMMENT a new inventory of sufficient size}, {@linkplain ArrayPropertyDelegate
+	COMMENT a new property delegate}, and {@linkplain ScreenHandlerContext#EMPTY
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]:
```suggestion
	COMMENT a new simple inventory of sufficient size}, {@linkplain ArrayPropertyDelegate
	COMMENT a new array property delegate}, and {@linkplain ScreenHandlerContext#EMPTY
```

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 7]
```diff
@@ -1,4 +1,128 @@
...
+	COMMENT <p>The factory should create a new instance of a screen handler with the server-side
+	COMMENT constructor (one that takes inventories, etc). If the screen handler requires
+	COMMENT a property delegate or a context, create an instance and pass it here.
+	COMMENT
+	COMMENT <p>As long as the screen handler only uses the slots and properties, there should not
+	COMMENT be any need for external synchronization.
+	COMMENT
+	COMMENT <h3 id="interaction">Interaction</h3>
+	COMMENT <p>Screen handler interaction mainly involves "slot clicks" and "button clicks".
+	COMMENT {@linkplain #onSlotClick Slot click} is, as mentioned before, an action manipulating
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]:
```suggestion
	COMMENT A {@linkplain #onSlotClick slot click} is, as mentioned before, an action manipulating
```

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 8]
```diff
@@ -118,25 +273,73 @@ CLASS net/minecraft/class_1703 net/minecraft/screen/ScreenHandler
 	METHOD method_7594 unpackQuickCraftStage (I)I
 		ARG 0 quickCraftData
 	METHOD method_7595 close (Lnet/minecraft/class_1657;)V
+		COMMENT Closes this screen handler.
+		COMMENT
+		COMMENT <p>To close a screen handler, call {@link PlayerEntity#closeHandledScreen}
+		COMMENT on the server instead of this method.
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandler.mapping:279]: So is this more of an "onClosed" 🤔 (renames can be another PR)

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 9]
```diff
@@ -145,19 +348,38 @@ CLASS net/minecraft/class_1703 net/minecraft/screen/ScreenHandler
 	METHOD method_7608 calculateComparatorOutput (Lnet/minecraft/class_2586;)I
 		ARG 0 entity
 	METHOD method_7609 onContentChanged (Lnet/minecraft/class_1263;)V
+		COMMENT Called when a slot's content has changed.
+		COMMENT
+		COMMENT <p>This is not called by default; subclasses that override this method
+		COMMENT should also use a custom {@link Inventory}  whose {@code markDirty} method is
```
Comment (Juuxel) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]:
```suggestion
		COMMENT should also use a custom {@link Inventory} whose {@link Inventory#markDirty markDirty} method is
```

**Review by Juuxel — CHANGES_REQUESTED** [review]
Great work! We should probably also get #2944 in at some point since the current name is pretty bad.

also, here's a screen handler :tiny_potato:  
a real screen handler

File: mappings/net/minecraft/screen/ScreenHandler.mapping (hunk 4 above)
Comment (apple502j) [reply mappings/net/minecraft/screen/ScreenHandler.mapping]: This takes a single `PropertyDelegate`, a list of properties that are delegated. So the word "delegate" is singular. (In the same way a list is named "list" not "lists")

Comment (Juuxel) [reply mappings/net/minecraft/screen/ScreenHandler.mapping]: Right, but screen handlers can have multiple property delegates (or none) if they call that method multiple times with different delegates. The current wording implies that all screen handlers will have exactly one PD, which isn't true.

**Review by modmuss50 — APPROVED** [review]
